### PR TITLE
Added extra versions in OPAM

### PR DESCRIPTION
**Pull request by scemama**

### CODE REVIEW

File: var/spack/repos/builtin/packages/opam/package.py [hunk 1]
```diff
@@ -16,19 +16,37 @@ class Opam(AutotoolsPackage):
...
+    @when('@:1.2.2')
     def build(self, spec, prefix):
         make('lib-ext')
         make()
         make('man')
+
+    @when('@2.0.0:')
+    def build(self, spec, prefix):
+        make('lib-ext')
+        make()
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/opam/package.py]: I would definitely avoid duplicating the `build` function, just do:
```python
def build(self, spec, prefix):
    make('lib-ext')
    make()
    if spec.satisfies('@:1.2.2'):
        make('man')
```

File: var/spack/repos/builtin/packages/opam/package.py [hunk 2]
```diff
@@ -16,19 +16,37 @@ class Opam(AutotoolsPackage):
...
     version('1.2.1', sha256='f210ece7a2def34b486c9ccfb75de8febd64487b2ea4a14a7fa0358f37eacc3b')
 
-    depends_on('ocaml')  # Not a strict dependency, but recommended
+    depends_on('ocaml')
 
     parallel = False
 
+    sanity_check_is_file = ['bin/opam']
+
+    @when('@:1.2.2')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/opam/package.py]: Can you explain in a brief comment why `OCAMLPARAM` was needed before and is not needed now?

Comment (scemama) [reply var/spack/repos/builtin/packages/opam/package.py]: The handling of strings has changed in OCaml, similarly to Python2/Python3 (differentiating strings and byte sequences).
Before OCaml 4.06.1, strings were mutable. Starting from OCaml 4.06.1, strings are now immutable. There has been a slow transition from strings to bytes from 4.02 to 4.06.1.

OPAM version 2 was written after OCaml 4.06.1 was released, so the code of OPAM version 2 considers immutable strings, and doesn't require the `safe-string=0` flag.

File: var/spack/repos/builtin/packages/opam/package.py (hunk 1 above)
Comment (scemama) [reply var/spack/repos/builtin/packages/opam/package.py]: OK thanks. Done.

File: var/spack/repos/builtin/packages/opam/package.py (hunk 2 above)
Comment (alalazo) [reply var/spack/repos/builtin/packages/opam/package.py]: Apologies. I meant if you could add a brief comment in the code, for future reference of anybody who'll modify the recipe :slightly_smiling_face:

**Pull request merged**